### DISCUSSION TITLE
Precautions when using NAME.surge.sh

### DISCUSSION THREAD

**Discussion by fu-sen** [sintaxi/surge#443] [Ideas]
**You should not use `NAME.surge.sh` for public websites.**

It never appears in search results. Some have been transferred to others, so Google will show them:
https://www.google.com/search?q=site%3Asurge.sh

`NAME.surge.sh/robots.txt` has this text fixed when you generate `NAME.surge.sh`:
```
User-agent: *
Disallow: /
```

Therefore, you should always use Custom Domain when publishing the Web with Surge.

**Comment by sintaxi** [comment]
Surge originally allowed `NAME.surge.sh` to be indexed by search engines and it provided an incentive for link farming to happen on the surge.sh subdomains. It would be nice if we can get those subdomains de-indexed and perhaps find a way to whitelist friendly users of surge for indexing on the surge.sh domain.

**Reply by fu-sen** [reply]
Yeah. It looks like you are suffering from some account malicious activity. It's also visible in the Google search results I linked to the day before (Is it an SEO measure? It looks like a wasteful act...😫) At least this account will need to be addressed. It runs Surge's server in vain.
Some users still know the existence of Freenom. I use [EU.org](https://nic.eu.org/) instead. It's a workaround in a sense, but it shouldn't be used for public web apps.

However, even if that is resolved, the ccTLD adopted by `surge.sh` is not desired. It's that some web services have changed their domain to gTLDs. Considering SSL, other work may be needed.

**Reply by fu-sen** [reply]
Malicious sites appear to be mechanically auto-generated, so limiting Student (Free Plan) builds may help with this issue. Accounts with many deployments in a short amount of time should obviously be suspended. Or limit simultaneous builds. This should have little effect on accounts that manually run `surge`. Some users may complain about this, but it's an account that should be upgraded to Surge Professional or decommissioned.

**Comment by fu-sen** [comment]
You've found that you've had problems with .pdf files in the past and are taking steps to whitelist by registering credit card:
https://github.com/sintaxi/surge/issues/257

Can this method be applied to `SUBDOMAIN.surge.sh` as well? I feel like it's ideal.
